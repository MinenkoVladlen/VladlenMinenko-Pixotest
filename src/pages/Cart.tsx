import {
    Container,
    Typography,
    List,
    ListItem,
    ListItemAvatar,
    Avatar,
    ListItemText,
    IconButton,
    Button,
    Grid2,
} from '@mui/material'
import DeleteIcon from '@mui/icons-material/Delete'
import ArrowBackIcon from '@mui/icons-material/ArrowBack'
import { useAppDispatch, useAppSelector } from '../hooks'
import { removeFromCart } from '../features/cart/cartSlice'
import { useNavigate } from 'react-router-dom'

const Cart: React.FC = () => {
    const cartItems = useAppSelector(state => state.cart.items)
    const dispatch = useAppDispatch()
    const navigate = useNavigate()

    const handleRemove = (id: number) => {
        dispatch(removeFromCart(id))
    }

    const total = cartItems.reduce((sum, item) => sum + item.price * item.quantity, 0)

    return (
        <Container sx={{ mt: 4, minHeight: '100vh' }}>
            <Button startIcon={<ArrowBackIcon />} onClick={() => navigate(-1)} sx={{ mb: 2 }}>
                Back
            </Button>
            <Typography variant="h4" gutterBottom>
                Cart
            </Typography>
            {cartItems.length === 0 ? (
                <Grid2 container sx={{ justifyContent: 'center' }}>
                    <Typography variant="h6">Your cart is empty </Typography>
                </Grid2>
            ) : (
                <>
                    <List>
                        {cartItems.map(item => (
                            <ListItem key={item.id} divider>
                                <ListItemAvatar>
                                    <Avatar src={item.image} variant="square" />
                                </ListItemAvatar>
                                <ListItemText
                                    primary={`${item.title}`}
                                    secondary={`Count: ${item.quantity} | Price: $${item.price}`}
                                />
                                <IconButton edge="end" onClick={() => handleRemove(item.id)}>
                                    <DeleteIcon />
                                </IconButton>
                            </ListItem>
                        ))}
                    </List>
                    <Typography variant="h6">Total: ${total.toFixed(2)}</Typography>
                </>
            )}
        </Container>
    )
}

export default Cart
