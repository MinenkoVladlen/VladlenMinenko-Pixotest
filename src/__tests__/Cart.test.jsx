import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { store } from '../store';
import Cart from '../pages/Cart';

describe('Cart Snapshot', () => {
    it('should match snapshot', () => {
        const { asFragment } = render(
            <Provider store={store}>
                <MemoryRouter>
                    <Cart />
                </MemoryRouter>
            </Provider>
        );

        expect(asFragment()).toMatchSnapshot();
    });
});