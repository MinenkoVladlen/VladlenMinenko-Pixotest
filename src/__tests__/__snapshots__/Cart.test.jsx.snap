// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart Snapshot should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-3ke3t9-MuiContainer-root"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-sv1a86-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeMedium css-1sh91j5-MuiButton-startIcon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
          data-testid="ArrowBackIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20z"
          />
        </svg>
      </span>
      Back
    </button>
    <h4
      class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1it4a80-MuiTypography-root"
    >
      Cart
    </h4>
    <div
      class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row css-16fqu4o-MuiGrid2-root"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 css-1miy0lu-MuiTypography-root"
      >
        Your cart is empty 
      </h6>
    </div>
  </div>
</DocumentFragment>
`;
